mypy/applytype.py

```diff
@@ -38,8 +38,12 @@ def apply_generic_arguments(callable: CallableType, types: List[Type],
                     types[i] = value
                     break
             else:
-                msg.incompatible_typevar_value(callable, i + 1, type, context)
-
+                arg_strings = tuple(msg.format(arg_type).replace('"', '')
+                                    for arg_type in callable.arg_types)
```

Comment:
I think you should check ``arg_type`` for the presence of a constrained type variable here instead of string representation below.